vrpn_Connection.C

```diff
@@ -6564,7 +6562,7 @@ char *vrpn_copy_file_name(const char *filespecifier)
     filename = NULL;
     try {
       filename = new char[len];
-      strncpy(filename, fp, len - 1);
+      strncpy(filename, fp, len);
       filename[len - 1] = 0;
```

Comment:
Since the code uses strlen() to find the length and then adds 1 to it, I guess this will always include the trailing 0.  In that case, doing the full-length copy makes sense, but then the following line setting the last character to 0 is no longer needed.  This must have been copied from code that worked even with a non-null-terminated string.

Reply:
So this and the other ones were all fixes to warnings with clang-8. I hadn't seen them before so I might have goofed them up, but strncpy is notoriously prickly - see the article linked in the comment to vrpn_strcpy.

```
../vrpn_Connection.C: In function ‘char* vrpn_copy_file_name(const char*)’:
../vrpn_Connection.C:6567:14: warning: ‘char* strncpy(char*, const char*, size_t)’ output truncated before terminating nul copying as many bytes from a string as its length [-Wstringop-truncation]
       strncpy(filename, fp, len - 1);
       ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
../vrpn_Connection.C:6563:21: note: length computed here
     len = 1 + strlen(fp);
               ~~~~~~^~~~
```

Reply:
It is true that this stops copying before the null character, but then the next line fills in the null character.  The full-length copy is better, but also makes the following line superfluous.